test/origin_simulator/body_test.exs

```diff
@@ -0,0 +1,36 @@
+defmodule OriginSimulator.RandomiserTest do
+  use ExUnit.Case
+
+  alias OriginSimulator.Body
+
+  describe "parsing a string containg a placeholder" do
+    test "keeps text outside of placeholders " do
+      decoded_body = Body.parse("{\"data\":\"<<1kb>>\"}") |> Poison.decode!()
+      assert String.length(decoded_body["data"]) == 1024
+    end
+
+    test "replaces placeholders with random content" do
+      parsed_string = Body.parse("{\"data\":\"some random<<4kb>>and also<<10kb>>this\"}")
+      refute parsed_string =~ ~r"<<.+?>>"
+    end
+  end
+
+  describe "parsing a string containg multiple placeholders" do
+    test "keeps text outside of placeholders " do
+      parsed_string = Body.parse("{\"data\":\"some random<<4kb>>and also<<10kb>>this\"}")
+      assert parsed_string =~ "some random"
```

Comment:
We don't know if this didn't replace anything. This could have ended up as the input as the beginning should always match? All we can guarantee is something was returned.

Reply:
Yeah, we could also assert that the size is > 14kb